src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.Types.cs

```diff
@@ -12,20 +12,32 @@ internal static partial class JSHostImplementation
 
         public sealed class PromiseHolder
         {
-            public nint GCVHandle;
+            public nint GCHandle;
             public ToManagedCallback? Callback;
 #if FEATURE_WASM_THREADS
             // the JavaScript object could only exist on the single web worker and can't migrate to other workers
-            internal int OwnerThreadId;
-            internal SynchronizationContext? SynchronizationContext;
+            internal JSSynchronizationContext SynchronizationContext;
+#endif
+
+#if FEATURE_WASM_THREADS
+            public PromiseHolder(JSSynchronizationContext targetContext)
```

Comment:
I feel like having a different signature for this per-ifdef is going to cause an explosion of ifdef'd code everywhere, when we could just have CurrentJSSynchronizationContext exist and return null